Aquarius/Src/Aquarius/Events/EventHandler.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "Event.h"
+
+#include <GLFW/glfw3.h>
+#include <unordered_map>
+#include <vector>
+
+
+namespace Aquarius {
+
+    template <typename T>
+    class EventHandler
+    {
+    public:
+        using functionType = std::function<void(const Event<T>&)>;
+        void subscribe(T Type, const functionType& function)
+        {
+            subscribers[Type].push_back(function);
+        }
+
+        void notify(Event<T>& event)
+        {
+            // Check to see if there are any subscribers of event type
+            if((subscribers.find(event.getEventType())) == subscribers.end())
+                return;
+
+            for(auto&& subscriber : subscribers.at(event.getEventType()))
+            {
+                // Notify of event only if the event has not already been handled
+                if(!event.isHandled())
+                {
+                    subscriber(event);
+                }
```

Comment:
So to me, this looks like a client can set the handled member variable to true, and then nothing else occurs. If the event is handled, does it make sense to just break out of this for loop? This will just loop through the rest of the vector and do nothing if the event has already been handled by one of the other subscribers

Reply:
@Burke-Daniel I think we should break when it gets handled for sure. However, some events may need information from an event but not actually consume the event, therefore we shouldnt break after line 43. Maybe we add a check after line 43, if (handled) {break;}